doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    collision_object.header.frame_id = move_group.getPlanningFrame();
+
+    // The id of the object is used to identify it
+    collision_object.id = "box2";
+
+    // Define a box to add to the world
+    shape_msgs::SolidPrimitive primitive;
+    primitive.type = primitive.BOX;
+    primitive.dimensions.resize(3);
+    primitive.dimensions[0] = 0.1;
+    primitive.dimensions[1] = 0.1;
+    primitive.dimensions[2] = 0.1;
+
+    // Define a pose for the box (specified relative to frame_id)
+    geometry_msgs::Pose box_pose;
+    box_pose.orientation.w = 1.0;
+    box_pose.position.x = 0.5;
+    box_pose.position.y = -0.2;
+    box_pose.position.z = 0.6;
+
+    collision_object.primitives.push_back(primitive);
+    collision_object.primitive_poses.push_back(box_pose);
+    collision_object.operation = collision_object.ADD;
+
+    collision_objects.push_back(collision_object);
+    object_ids.push_back(collision_object.id);
+
+    // Add the collision object into the world
+    ROS_INFO_NAMED("tutorial", "Adding an object into the world");
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Attach Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+
+    planning_scene_interface.addCollisionObjects(collision_objects);
+  }
+
+  void attach_parcel()
+  {
+    visual_tools.prompt("Press 'next' to attach the object");
+
+    // BEGIN_SUB_TUTORIAL attach_obj
+    // Attaching and detaching objects
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // Attaching or detaching an already existing object is easy.
+    //
+    // To attach an object.
+    move_group.attachObject(collision_objects.back().id);
+    // END_SUB_TUTORIAL
+  }
+
+  void plan_parcel()
+  {
+    // Rviz
+    visual_tools.prompt("Press 'next' to plan and move with the attached object");
+
+    // target_pose repositioning and planning
+    target_pose.position.y += 0.2;
+    target_pose.position.z += 0.2;
+    move_group.setPoseTarget(target_pose);
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 7 (pose goal with attached cuboid) %s", success ? "" : "FAILED");
+
+    // Rviz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishAxisLabeled(target_pose, "point");
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.publishText(text_pose, "Pose Goal with Attached Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+  }
+
+  void detach_parcel()
+  {
+    ROS_INFO_NAMED("tutorial", "Detach the object from the robot");
+
+    // BEGIN_SUB_TUTORIAL detach_obj
+    // To detach an object.
+    move_group.detachObject(collision_objects.back().id);
+    // END_SUB_TUTORIAL
+
+    // RViz
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Object Detached", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+  }
+
+  void remove_parcel()
```

Comment:
pass in the id. Check if the object is in the planning_scene first. Return true if it was found and removed

Reply:
done